src/dev/flang/parser/Parser.java

```diff
@@ -2794,10 +2794,13 @@ If ifexpr()
   {
     return relaxLineAndSpaceLimit(() -> {
         SourcePosition pos = tokenSourcePos();
+        // set new min indent unless this `if` is part of an `else if`
+        var old = lastToken() == Token.t_else ? null : setMinIndent(tokenPos());
         match(Token.t_if, "ifexpr");
         Expr e = exprInLine();
         Block b = thenPart(false);
         var els = elseBlock();
+        if (old != null) { setMinIndent(old); }
```

Comment:
This will omit the `setMinIndent` if used in a precondition `pre else` like
```
  f
  pre else if cc then true 
  else false
  is ...
```
Same problem if used in a loop's `else`
```
  while cc1
  else if cc2 then say "a" 
  else say "b"
```
thse should better be required to be
```
  f
  pre else if cc then true 
           else false
  is ...
```
and
```
  while cc1
  else if cc2 then say "a" 
       else say "b"
```
Maybe we need to change the grammar for `elseBlock` to check if fallowed by `if` and if this is the case and the `elseBlock` is used as part of an `if`, only then `setMIntIndent` should be skipped.